### PR TITLE
review comments

### PR DESCRIPTION
**Pull request by ckamm**

### CODE REVIEW

File: programs/dasheri/src/iou/instructions/init_gateway.rs [hunk 1]
```diff
@@ -13,6 +13,14 @@ pub struct IouInitGateway<'info> {
     )]
     pub gateway: Box<Account<'info, Gateway>>,
 
+    // REVIEW: I think there's a nicer way to init a new mint:
```
Comment (microwavedcola1) [inline programs/dasheri/src/iou/instructions/init_gateway.rs:16]: done

File: programs/dasheri/src/pool/instructions/deposit_into_pool.rs [hunk 2]
```diff
@@ -49,6 +49,9 @@ impl<'info> PoolDepositIntoPool<'info> {
     }
 }
 
+// REVIEW: What's this instruction good for? Why can I deposit into the vault?
```
Comment (microwavedcola1) [inline programs/dasheri/src/pool/instructions/deposit_into_pool.rs:52]: users would deposit into the pool, and a pool admin would deposit it into the pools mango account

File: programs/dasheri/src/pool/instructions/create_mango_account.rs [hunk 3]
```diff
@@ -6,9 +6,15 @@ use solana_program::program::invoke_signed;
 #[derive(Accounts)]
 #[instruction(account_num: u64, bump: u8)]
 pub struct PoolCreateMangoAccount<'info> {
+    // REVIEW: So this can be done for any mango program/group permissionlessly?
```
Comment (microwavedcola1) [inline programs/dasheri/src/pool/instructions/create_mango_account.rs:9]: one can definitely hard code the address, I will add a todo

Comment (microwavedcola1) [reply programs/dasheri/src/pool/instructions/create_mango_account.rs:9]: do you think any other check then target address would be useful?

File: programs/dasheri/src/pool/instructions/create_mango_account.rs [hunk 4]
```diff
@@ -6,9 +6,15 @@ use solana_program::program::invoke_signed;
...
+    // There's probably no way to validate that mango_program is a real mango program...
```
Comment (microwavedcola1) [inline programs/dasheri/src/pool/instructions/create_mango_account.rs:10]: at least a program check could be done, I think only programs are marked as executable no?

File: programs/dasheri/src/pool/instructions/create_mango_account.rs [hunk 5]
```diff
@@ -6,9 +6,15 @@ use solana_program::program::invoke_signed;
...
     pub mango_program: AccountInfo<'info>,
+
     #[account(mut)]
     pub mango_group: AccountInfo<'info>,
+
+    // REVIEW: I'd put seed constraints here, since it's a PDA
+    // maybe unnecessary since the mango program will duplicate the check
```
Comment (microwavedcola1) [inline programs/dasheri/src/pool/instructions/create_mango_account.rs:17]: yup, check is done in mango

Comment (microwavedcola1) [reply programs/dasheri/src/pool/instructions/create_mango_account.rs:17]: but I will add a note

File: programs/dasheri/src/pool/instructions/create_pool.rs [hunk 6]
```diff
@@ -29,6 +29,7 @@ pub struct PoolCreatePool<'info> {
     // )]
     pub deposit_iou_mint: Box<Account<'info, Mint>>,
 
+    // REVIEW: in prod you'd likely have a different admin and pda-payer
```
Comment (microwavedcola1) [inline programs/dasheri/src/pool/instructions/create_pool.rs:32]: agreed, I have been lazy with separating the two

File: programs/dasheri/src/pool/instructions/create_pool_account.rs [hunk 7]
```diff
@@ -7,6 +7,7 @@ use crate::pool::state::{Pool, PoolAccount};
 pub struct PoolCreatePoolAccount<'info> {
     #[account(
         init,
+        // REVIEW: Using these seeds, there can be only one pool_account per pool!
```
Comment (microwavedcola1) [inline programs/dasheri/src/pool/instructions/create_pool_account.rs:10]: fixed

File: programs/dasheri/src/pool/instructions/create_pool_account.rs [hunk 8]
```diff
@@ -19,6 +20,7 @@ pub struct PoolCreatePoolAccount<'info> {
     )]
     pub pool: Box<Account<'info, Pool>>,
 
+    // REVIEW: possibly pool account owner and payer would be different
```
Comment (microwavedcola1) [inline programs/dasheri/src/pool/instructions/create_pool_account.rs:23]: again yes I have been lazy with these

File: programs/dasheri/src/pool/instructions/create_pool.rs (hunk 6 above)
Comment (microwavedcola1) [reply programs/dasheri/src/pool/instructions/create_pool.rs:32]: added todo

File: programs/dasheri/src/pool/instructions/create_pool_account.rs (hunk 8 above)
Comment (microwavedcola1) [reply programs/dasheri/src/pool/instructions/create_pool_account.rs:23]: added todo

File: programs/dasheri/src/pool/instructions/create_mango_account.rs [hunk 9]
```diff
@@ -25,6 +31,8 @@ pub struct PoolCreateMangoAccount<'info> {
     pub system_program: Program<'info, System>,
 }
 
+// REVIEW: so there could be multiple mango_accounts per pool?
```
Comment (microwavedcola1) [inline programs/dasheri/src/pool/instructions/create_mango_account.rs:34]: technically yes, you can use the account_num to do that, of course if you use the same pda it would crash, I would prefer keeping it separate, since the number of accounts going to the ix blows up and becomes hard to read (did try it just now and got really hard to follow)

File: programs/dasheri/src/pool/instructions/create_mango_account.rs (hunk 3 above)
Comment (ckamm) [reply programs/dasheri/src/pool/instructions/create_mango_account.rs:9]: no, I expect it is fine to be able to use this program with any mango program/group; definitely fine for the example

File: programs/dasheri/src/pool/instructions/create_mango_account.rs (hunk 4 above)
Comment (ckamm) [reply programs/dasheri/src/pool/instructions/create_mango_account.rs:10]: Yeah, but I meant that a sneaky attacker program could provide the API your program needs, but then run away with the funds. But it's up to the caller of this function here to enter the correct ids, should not be an issue
